foundation-models/openai/src/main/java/com/sap/ai/sdk/foundationmodels/openai/OpenAiSystemMessage.java

```diff
@@ -15,23 +24,69 @@
 @Beta
 @Value
 @Accessors(fluent = true)
-class OpenAiSystemMessage implements OpenAiMessage {
+@AllArgsConstructor(access = PACKAGE)
+public class OpenAiSystemMessage implements OpenAiMessage {
 
-  /** The role of the message. */
+  /** The role associated with this message. */
   @Nonnull String role = "system";
 
   /** The content of the message. */
-  @Nonnull String content;
+  @Getter(onMethod_ = @Beta)
+  @Nonnull
+  OpenAiMessageContent content;
+
+  /**
+   * Creates a new system message from a string.
+   *
+   * @param message the first message.
+   */
+  @Tolerate
+  OpenAiSystemMessage(@Nonnull final String message) {
+    this(new OpenAiMessageContent(List.of(new OpenAiTextItem(message))));
+  }
+
+  /**
+   * Add text to the message.
+   *
+   * @param message the text to add.
+   * @return the new message.
+   */
+  @Nonnull
+  public OpenAiSystemMessage withText(@Nonnull final String message) {
+    final var contentItems = new LinkedList<>(content.items());
+    contentItems.add(new OpenAiTextItem(message));
+    return new OpenAiSystemMessage(new OpenAiMessageContent(contentItems));
+  }
 
   /**
    * Converts the message to a serializable object.
    *
    * @return the corresponding {@code ChatCompletionRequestSystemMessage} object.
    */
   @Nonnull
-  ChatCompletionRequestSystemMessage createChatCompletionRequestMessage() {
+  ChatCompletionRequestSystemMessage createChatCompletionRequestMessage()
+      throws IllegalArgumentException {
+    final var itemList = this.content().items();
+    if (itemList.size() == 1 && itemList.get(0) instanceof OpenAiTextItem textItem) {
+      return new ChatCompletionRequestSystemMessage()
+          .role(ChatCompletionRequestSystemMessage.RoleEnum.fromValue(role()))
+          .content(ChatCompletionRequestSystemMessageContent.create(textItem.text()));
+    }
+
+    final var contentList = new LinkedList<ChatCompletionRequestSystemMessageContentPart>();
+    for (final OpenAiContentItem item : itemList) {
+      if (item instanceof OpenAiTextItem textItem) {
+        contentList.add(
+            new ChatCompletionRequestMessageContentPartText()
+                .type(ChatCompletionRequestMessageContentPartText.TypeEnum.TEXT)
+                .text(textItem.text()));
+      } else {
+        final var errorMessage = "Unknown content type for " + role() + " messages.";
```

Comment:
```suggestion
        final var errorMessage = "Unknown content type " + item.getClass() + " for " + role() + " messages.";
```
something like that, also same for the other message types